### PR TITLE
 Exclude room-specific commands for other rooms in /permissions

### PR DESCRIPTION
**Pull request by mia-pi-git**
Does what it says on the tin. This is much easier now with `this.requireRoom(roomid)`.

### CODE REVIEW

File: server/chat.ts [hunk 1]
```diff
@@ -1720,7 +1721,8 @@ export const Chat = new class {
 			if (typeof entry !== 'function') continue;
 
 			const handlerCode = entry.toString();
-			entry.requiresRoom = /\bthis\.requires?Room\(/.test(handlerCode);
+			entry.requiresRoom = /\bthis\.require?Room\(/.test(handlerCode);
+			entry.roomSpecific = /requireRoom\((?:'|"|`)(.*?)(?:'|"|`).\)/.exec(handlerCode)?.[1];
```
Comment (Zarel) [inline server/chat.ts:1725]: Why not just turn `requiresRoom` into `RoomID | null`? I don't see any backwards compatibility issues.

Comment (mia-pi-git) [reply server/chat.ts]: Done.

File: server/chat.ts [hunk 2]
```diff
@@ -1720,7 +1720,7 @@ export const Chat = new class {
 			if (typeof entry !== 'function') continue;
 
 			const handlerCode = entry.toString();
-			entry.requiresRoom = /\bthis\.requires?Room\(/.test(handlerCode);
+			entry.requiresRoom = /requireRoom\((?:'|"|`)(.*?)(?:'|"|`).\)/.exec(handlerCode)?.[1] || false;
```
Comment (Zarel) [inline server/chat.ts:1723]: Will this be `true` for `requireRoom()` with no arguments?

Comment (Zarel) [reply server/chat.ts:1723]: Looks like no. That'll need to be fixed.

Comment (mia-pi-git) [reply server/chat.ts]: Fixed.

File: server/chat.ts [hunk 3]
```diff
@@ -1720,7 +1720,7 @@ export const Chat = new class {
 			if (typeof entry !== 'function') continue;
 
 			const handlerCode = entry.toString();
-			entry.requiresRoom = /\bthis\.requires?Room\(/.test(handlerCode);
+			entry.requiresRoom = /requireRoom\((?:'|"|`)(.*?)(?:'|"|`).\)/.exec(handlerCode)?.[1] || /this\.requireRoom\(/.test(handlerCode);
```
Comment (Zarel) [inline server/chat.ts:1723]: `entry.requiresRoom` seems to still be a `boolean`? That should be updated to `boolean | RoomID`, right?

Comment (Zarel) [inline server/chat.ts:1723]: What's with the `.\)`? There should be a cast to `RoomID` after the string, right? There's no reason to search for the closing paren anyway, right?

File: server/chat.ts [hunk 4]
```diff
@@ -1720,7 +1720,7 @@ export const Chat = new class {
 			if (typeof entry !== 'function') continue;
 
 			const handlerCode = entry.toString();
-			entry.requiresRoom = /\bthis\.requires?Room\(/.test(handlerCode);
+			entry.requiresRoom = /requireRoom\((?:'|"|`)(.*?)(?:'|"|`) /.exec(handlerCode)?.[1] as RoomID || /this\.requireRoom\(/.test(handlerCode);
```
Comment (Zarel) [inline server/chat.ts:1723]: Why space?
```suggestion
			entry.requiresRoom = /requireRoom\((?:'|"|`)(.*?)(?:'|"|`)/.exec(handlerCode)?.[1] as RoomID || /this\.requireRoom\(/.test(handlerCode);
```

**Comment by Zarel** [comment]
Build failure seems unrelated to this PR; merging so I don't forget.

**Pull request merged**